### PR TITLE
Fix ReactDomClient bug

### PR DESCRIPTION
**Pull request by Judahmeek**
### Summary

_Remove this paragraph and provide a general description of the code changes in your pull
request... were there any bugs you had fixed? If so, mention them. If
these bugs have open GitHub issues, be sure to tag them here as well,
to keep the conversation linked together._

### Other Information

_Remove this paragraph and mention any other important and relevant information such as benchmarks._

### Pull Request checklist
_Remove this line after checking all the items here. If the item is not applicable to the PR, both check it out and wrap it by `~`._

- [ ] Add/update test to cover these changes
- [ ] Update documentation
- [ ] Update CHANGELOG file

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -6,17 +6,33 @@
...
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler: 2.4.9
+          ruby-version: ${{ matrix.ruby }}
+      - uses: actions/setup-node@v3
+      - run: bundle install
+      - run: yarn
+      - run: bundle exec rake react:update
+      - run: bundle exec rake ujs:update
+      - run: ./check_for_uncommitted_files.sh
```
Comment (ahangarha) [inline .github/workflows/ruby.yml:29]: I think it is better to extract this logic into another rake task and use it in both:
- Release script
- CI

Is will be DRY and also prevents adding bash script in the root of the project.

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -6,17 +6,33 @@
...
   test:
     strategy:
       fail-fast: false
       matrix:
-        ruby:
-          - 2.7.0
-          - 2.6.5
-          # TODO restore testing against JRuby w/ therubyrhino
-          # - jruby-9.3.7.0
+        ruby: [2.7]
```
Comment (ahangarha) [inline .github/workflows/ruby.yml:36]: Not related to this PR, but I think we should add ruby 3 to the matrix for our tests.

File: rakelib/create_release.rake [hunk 3]
```diff
@@ -39,7 +39,7 @@ task :create_release, %i[gem_version dry_run] do |_t, args|
   puts 'Updating ujs:update'
   Rake::Task['ujs:update'].invoke
 
-  Release.commit_the_changes('Update pre-bundled react and React ujs') unless is_dry_run
+  Release.make_sure_react_and_ujs_are_updated_and_commited
```
Comment (ahangarha) [inline rakelib/create_release.rake:42]: With this change, we don't need the `commit_the_changes` method anymore.

File: rakelib/create_release.rake [hunk 4]
```diff
@@ -87,6 +87,19 @@ module Release
...
+      return if $CHILD_STATUS.success? && status == ''
+
+      error = if $CHILD_STATUS.success?
+                'Running react:update and ujs:update resulted in uncommitted changes. Please commit those changes & confirm that CI passes before continuing.'
+              else
+                'You do not have Git installed. Please install Git, and commit your changes before continuing'
+              end
+      raise(error)
+    end
+
```
Comment (ahangarha) [inline rakelib/create_release.rake:102]: As mentioned before, this can become a separate rake task

**Review by ahangarha — COMMENTED** [review]
Thanks for the improvements 👍🏾 

I have made some comments for refactoring.

File: .github/workflows/ruby.yml (hunk 2 above)
Comment (justin808) [reply .github/workflows/ruby.yml:36]: followup PR

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (justin808) [reply .github/workflows/ruby.yml:29]: I'm not convinced.

File: react_ujs/src/reactDomClient.js [hunk 5]
```diff
@@ -1,24 +1,25 @@
...
+    } catch (e) {
+      // We should never get here, but if we do, we'll just use the default ReactDOM
+      // and live with the warning.
+      reactDomClient = ReactDOM;
+    }
   }
 }
 
-export default reactDomClient
+export default reactDomClient
```
Comment (justin808) [inline react_ujs/src/reactDomClient.js]: @Judahmeek set your edit to add the extra line at the end.

**Review by justin808 — APPROVED** [review]

**Comment by justin808** [comment]
@Judahmeek we're getting a CI failure: https://github.com/reactjs/react-rails/actions/runs/5020783720/jobs/9002562807?pr=1278

**Pull request merged**